src/main/java/com/artipie/maven/util/AutoCloseablePath.java

```
@@ -0,0 +1,120 @@
+
+/**
+ * Adapts {@link Path} to {@link AutoCloseable} interface, quietly deleting on close.
+ * @since 0.1
+ */
+public final class AutoCloseablePath implements AutoCloseable {
+
+    /**
+     * Class logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(AutoCloseablePath.class);
+
+    /**
+     * Actual path.
+     */
+    private final Path path;
+
+    /**
+     * All arg constructor.
+     * @param path A path to instance to wrap over.
+     */
+    public AutoCloseablePath(final Path path) {
+        this.path = path;
+    }
+
+    @Override
+    public void close() throws Exception {
+        try {
+            Files.deleteIfExists(this.path);
+        } catch (final IOException ex) {
+            LOG.warn(String.format("with %s", this.path), ex);
+        }
+    }
+
+    /**
+     * Returns internal Path instance.
+     * @return Path instance
+     */
+    public Path unwrap() {
+        return this.path;
+    }
+
+    /**
+     * Encapsulates parent directory logic to not to expose it as a raw {@link Path}.
+     * @since 0.1
+     */
+    public static class Parent {
+
+        /**
+         * Root directory.
+         */
+        private final Path dir;
+
+        /**
+         * All args constructor.
+         * @param dir Root directory
+         */
+        public Parent(final Path dir) {
+            this.dir = dir;
+        }
+
+        /**
+         * Resolves given path (if it's relative path) as a child of the root.
+         * Creates parent directories if needed
+         * @param path A child path.
+         * @return AutoCloseablePath instance
+         */
+        public AutoCloseablePath resolve(final Path path) {
+            try {
+                final var file = this.dir.resolve(path);
+                Files.createDirectories(file.getParent());
+                return new AutoCloseablePath(file);
+            } catch (final IOException ex) {
+                throw new UncheckedIOException(ex);
```

@Nimmaster why to hide exception with unchecked here?

---

It's not hiding, it's rethrowing